### ISSUE TITLE
DecimalFormat Does Not Suppress Negative Sign

### ISSUE THREAD

**Issue by mlvn23** [cquiroz/scala-java-locales#167]
This test case passes in JVM but not in JS:

```
        val uut2 = new DecimalFormat("0.00; 0.00")
        checkEquals("7.65", uut2.format(7.65))
        checkEquals(" 7.65", uut2.format(-7.65))
        checkEquals("98765.23", uut2.format(98765.234))
        checkEquals(" 98765.23", uut2.format(-98765.234))
```

**Comment by cquiroz** [comment]
Thanks for the report, I tested this and you're right. Are you interested on fixing it?

**Comment by mlvn23** [comment]
Apparently, this also does not work:

```
        val uut0 = new DecimalFormat("000")
        checkEquals("001", uut0.format(1))
```

**Comment by mlvn23** [comment]
I looked at the JavaDoc and it does not look trivial to implement it from scratch. I saw a couple of implementation in JavaScript in Github. One is has MIT and another has BSD-3. Do you think it's okay to port it?

**Comment by cquiroz** [comment]
I agree it is not trivial, the algorithm was created by hand and obviously the parsing of the format is missing some cases. 
Unfortunately I don't have time to purse this
About porting please be mindful of the scala.js rules:
https://github.com/scala-js/scala-js/blob/master/DEVELOPING.md

Essentially you can't look at jdk code or harmony

**Comment by cquiroz** [comment]
I added your example as an ignored test
https://github.com/cquiroz/scala-java-locales/blob/master/testSuite/shared/src/test/scala/testsuite/javalib/text/DecimalFormatTest.scala#L674

**Comment by mlvn23** [comment]
If you're okay with an external JS lib, this implementation appears to be good:

https://github.com/DeloitteDigitalAPAC/DecimalFormat

They did it with a formalized [parser](https://github.com/DeloitteDigitalAPAC/DecimalFormat/blob/master/src/decimalformat.pegjs) and lots of tests. It also accepts a locale as input.

I did a browserify with it and ran it in ScalaJS with some of my unit test and so far so good. I'll try to do a CI test later this week to see if it holds up.

**Comment by cquiroz** [comment]
That's interesting. I wouldn't expect a js library implementing a java spec. It maybe used I think provided it doesn't add up too much in terms of size.
I want to support scala-native so this may need to go into a js specific version

**Comment by mlvn23** [comment]
I've been using it for two weeks trying different things and it seems solid. About 14KB minimized.

**Comment by cquiroz** [comment]
That sounds great, if you want to submit a PR it would be ok, Let's keep the old version in a cross project for a future scala native release
